### PR TITLE
TM 159 remaining funds fields

### PR DESCRIPTION
**Pull request by hasanrq**

### CODE REVIEW

**Review by chrisfoster76 — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.LevyTransferMatching.Api/Models/Pledges/GetApplicationsResponse.cs [hunk 1]
```diff
@@ -54,6 +56,8 @@ public class Application
             return new GetApplicationsResponse
             {
                 PledgeStatus = v.PledgeStatus,
+                PledgeRemainingAmount = v.RemainingAmount,
```
[the remark this answers is not in the document]
Comment (hasanrq) [reply src/SFA.DAS.LevyTransferMatching.Api/Models/Pledges/GetApplicationsResponse.cs:59]: I have added the new fields to unit test ... Its just basic checking as I don't think we can check that remaining amount is less than (or equal to) total amount.

File: src/SFA.DAS.LevyTransferMatching.Api.UnitTests/Controllers/PledgeTests/GetPledgeApplicationsTests.cs [hunk 2]
```diff
@@ -45,6 +45,8 @@ public async Task GetApplications_Returns_GetApplicationsResponse()
             Assert.IsNotNull(okObjectResult);
             Assert.IsNotNull(response);
             Assert.AreEqual(_queryResult.Applications.Count(), response.Applications.Count());
+            Assert.IsTrue(response.PledgeRemainingAmount >= 0);
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching.Api.UnitTests/Controllers/PledgeTests/GetPledgeApplicationsTests.cs:48]: We could assert equality with the corresponding source value in `_queryResult`.

**Review by chrisfoster76 — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.LevyTransferMatching.Api.UnitTests/Controllers/PledgeTests/GetPledgeApplicationsTests.cs (hunk 2 above)
Comment (SHESFA) [reply src/SFA.DAS.LevyTransferMatching.Api.UnitTests/Controllers/PledgeTests/GetPledgeApplicationsTests.cs]: Done now.

File: src/SFA.DAS.LevyTransferMatching.Api.UnitTests/Controllers/PledgeTests/GetPledgeApplicationsTests.cs [hunk 3]
```diff
@@ -45,8 +45,8 @@ public async Task GetApplications_Returns_GetApplicationsResponse()
             Assert.IsNotNull(okObjectResult);
             Assert.IsNotNull(response);
             Assert.AreEqual(_queryResult.Applications.Count(), response.Applications.Count());
-            Assert.IsTrue(response.PledgeRemainingAmount >= 0);
-            Assert.IsTrue(response.PledgeTotalAmount >= 0);
+            Assert.GreaterOrEqual(response.PledgeRemainingAmount, _queryResult.RemainingAmount);
+            Assert.GreaterOrEqual(response.PledgeTotalAmount, _queryResult.TotalAmount);
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching.Api.UnitTests/Controllers/PledgeTests/GetPledgeApplicationsTests.cs:49]: I don't quite follow - the values in _queryResult are valid if they are _greater_ than those in the response?

**Review by chrisfoster76 — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.LevyTransferMatching.Api.UnitTests/Controllers/PledgeTests/GetPledgeApplicationsTests.cs (hunk 3 above)
Comment (SHESFA) [reply src/SFA.DAS.LevyTransferMatching.Api.UnitTests/Controllers/PledgeTests/GetPledgeApplicationsTests.cs]: I think equal is the best test here.  Changes done.  Ready for review again.

**Review by chrisfoster76 — DISMISSED** [review]

**Review by chrisfoster76 — APPROVED** [review]

**Pull request merged**